### DISCUSSION TITLE
@types/js-cookie : Incompatible with js-cookie V3

### DISCUSSION THREAD

**Discussion by roddypratt** [DefinitelyTyped/DefinitelyTyped#54767] [Issues with a @types package]
From here https://github.com/js-cookie/js-cookie/issues/717

I've just tried upgrading to V3 and it looks like the DefinitelyTyped typings are out of step with the new module structure.

```
import * as Cookies from "js-cookie"
let test = Cookies.get("MyCookie")
```
...gives...
`export 'get' (imported as 'Cookies') was not found in 'js-cookie' (possible exports: default)`

**Comment by FagnerMartinsBrack** [comment]
Ping @theodorejb, contributor of types for js-cookie since https://github.com/js-cookie/js-cookie/issues/50. He might be able to help on this.

**Comment by carhartl** [comment]
If it helps, I'm in the process of finishing a full TypeScript variant of js-cookie here: https://github.com/carhartl/ts-cookie

From my point of view typings are complete:
https://github.com/carhartl/ts-cookie/blob/71935bf6e429f2886f084201863dddb3df543aaa/types/index.ts#L32-L47

Not sure how to export just these to make them available in DefinitelyTyped, or whether it will be obsolete anyway given a soon to be released ts-cookie package...

**Comment by carhartl** [comment]
Released https://github.com/carhartl/typescript-cookie on NPM, which provides fully compatible type definitions for js-cookie.

**Comment by xianshenglu** [comment]
Why not fixing in js-cookie instead of creating a new library? You know, it's risky to switch libraries. @carhartl
